src/Time/Units.hs

```diff
@@ -163,6 +169,15 @@ instance NFData (Time (rat :: Rat))
 instance Serialise (Time (rat :: Rat))
 #endif
 
+#ifdef HAS_aeson
+instance (KnownUnitName unit) => ToJSON (Time (unit :: Rat)) where
+    toJSON = toJSON . show
+
+instance (KnownUnitName unit) => FromJSON (Time (unit :: Rat)) where
+    parseJSON (String t) = pure $ read @(Time unit) $ Text.unpack t
```

Comment:
It's better to use `readMaybe` here and report error with `fail` function. Otherwise `read` will throw pure exception which we can't handle.

Also, instead of manual pattern-matching on `String` it's better to use [`withText`](https://hackage.haskell.org/package/aeson-1.2.4.0/docs/Data-Aeson.html#v:withText) function.